WordPress/gutenberg
# Link Control: persist advanced settings toggle state to preferences if available

This is an alternative to https://github.com/WordPress/gutenberg/pull/52321. Fixes #52216.



## What?
Once a user opens the advanced section of link control it will stay open until they close it again.

## Why?
This retains the UI state which will be useful for people who often use the "open in new tab" option.

## How?
Adds a new preference that gets toggled when you open the advanced section of link control.

## Testing Instructions
1. Add a paragraph
2. Add a link to some of the text
3. Open the advanced section of link control
4. Close the link editing UI
5. Reopen the link editing UI and confirm that the advanced section is open
6. Close the advanced option
7. Close the link editing UI
8. Reopen the link editing UI and confirm that the advanced section is still closed.
9. Also try reloading the edftor and checking the setting persists across reloads.
10. Also try in different editor environments and check the setting persists.

### Automated Tests

e2e tests  - `npm run test:e2e:playwright test/e2e/specs/editor/blocks/links.spec.js -- -g "toggle state of advanced link settings is preserved across editing links"`


Co-authored-by: Dave Smith <444434+getdave@users.noreply.github.com>
